src/app/shared/components/TopBar/SelectBoxes/index.tsx

```diff
@@ -0,0 +1,58 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { usePathname, useRouter } from 'next/navigation'
+
+import type { Sport, Team } from '@prisma/client'
+
+import { BASE_API_URL } from '@api/base'
+
+import SelectBox from './SelectBox'
+
+interface SelectSportProps {
+  sports: Sport[]
+}
+
+const SelectBoxes = (props: SelectSportProps) => {
+  const { sports } = props
+
+  const router = useRouter()
+
+  const pathname = usePathname()
+  const [_, selectedSportKey] = pathname.split('/')
+
+  const selectedSportId = sports.find(sport => sport.key === selectedSportKey)?.id
+
+  const [teams, setTeams] = useState<Team[]>([])
+
+  useEffect(() => {
+    if(selectedSportKey){
+      fetch(`${BASE_API_URL}/teams?sportId=${selectedSportId}`)
+        .then(response => response.json())
+        .then(teamsResponse => setTeams(teamsResponse))
+    }
+  }, [selectedSportKey])
+
+  const shouldShowTeamSelect = !!selectedSportKey && !!teams.length
+
+  return (
+    <div className="flex">
+      <SelectBox
+        callbacks={{ onChange: (e) => router.push(`/${e.target.value}`) }}
+        label="Sport"
+        selectOptions={sports}
+      />
+
+      {shouldShowTeamSelect && (
+        <SelectBox
+          callbacks={{ onChange: (e) => router.push(window.location.pathname + `/${e.target.value}`) }}
```

Comment:
we need to re-think this code, there is a bug here if you set this dropdown to Please Select and then back to Wallabies it just appends the value again so you can get `http://localhost:3000/rugby/australia/australia`

i think we need to be using `<Link />` from next to be changing the route, not `router.push`

Reply:
yeh that makes sense